Solution1/calculator_app/Program.cs

```diff
@@ -0,0 +1,119 @@
+// операции и вывести результат на экран. Поддерживаемые операции:
+// + - / * ^ (возведение в степень) 
+
+namespace CalculatorApp
+{
+    class Program
+    {
+        static bool TryToConvertInsertedOperandToInteger(string insertedOperand)
+        {
+            double integerOperand;
+            bool operandIsInteger = double.TryParse(insertedOperand, out integerOperand);
+            return operandIsInteger;
+        }
+
+        static double ConvertInsertedOperandToInteger(string insertedOperand)
+        {
+            double integerOperand;
+            bool operandIsInteger = double.TryParse(insertedOperand, out integerOperand);
+            return integerOperand;
+        }
+
+        static bool CheckIsOperationPossible(string insertedOperation)
+        {
+            bool isOperationPossible;
+            if (insertedOperation == "+" || insertedOperation == "-" || insertedOperation == "*" || insertedOperation == "/" || insertedOperation == "^")
+            {
+                isOperationPossible = true;
+            }
+            else
+            {
+                isOperationPossible = false;
+            }
+            return isOperationPossible;
+        }
+
+        static void Main()
+        {
+            do
+            {
+                bool isOperandInteger;
+                double firstOperand;
+                do
+                {
+                    Console.WriteLine("Enter first integer operand");
+                    string insertedFirstOperand = Console.ReadLine();
+
+                    isOperandInteger = TryToConvertInsertedOperandToInteger(insertedFirstOperand);
+                    firstOperand = ConvertInsertedOperandToInteger(insertedFirstOperand);
+
+                    if (isOperandInteger == false)
+                    {
+                        Console.WriteLine("You have entered not an integer!");
+                    }
+                }
+                while (isOperandInteger == false);
+
+                double secondOperand;
+                do
+                {
+                    Console.WriteLine("Enter second integer operand");
+                    string insertedSecondOperand = Console.ReadLine();
+
+                    isOperandInteger = TryToConvertInsertedOperandToInteger(insertedSecondOperand);
+                    secondOperand = ConvertInsertedOperandToInteger(insertedSecondOperand);
+
+                    if (isOperandInteger == false)
+                    {
+                        Console.WriteLine("You have entered not an integer!");
+                    }
+                }
+                while (isOperandInteger == false);
+
+                bool isOperationPossible;
+                do
+                {
+                    Console.WriteLine("What operation is necessary to carry out?");
+                    string operation = Console.ReadLine();
+
+                    isOperationPossible = CheckIsOperationPossible(operation);
```

Comment:
Extra operation, you could set isOperationPossible  in default statement of the switch-case and got default value as true for do-while cycle

Reply:
Done